.github/workflows/mdbook.yml

```diff
@@ -2,30 +2,21 @@ name: Publish to GitHub Pages
 
 on:
   push:
-    branches:
```

Comment:
@officialcjunior This is the reason it got deployed on PR. Need to restrict to push only on master.